### PR TITLE
Add typescript and javascript format.insertSpaceBeforeFunctionParenthesis

### PR DESCRIPTION
**Pull request by ChiriVulpes**
[Original Issue](https://github.com/Microsoft/vscode/issues/15386)

This issue was closed and hasn't been reopened since the feature 
[has been implemented in TypeScript](https://github.com/Microsoft/TypeScript/issues/12234).

Figured I would try my hand at completing it. I'm too busy at the moment to get a local build working, so I figured I'd just do the changes it looks like are necessary and then make a PR. If it's broken it can be rejected, and in that case I'll apologise for wasting your time. >.>

If you have time and this doesn't provide everything necessary, let me know. I'll try make a local build and update the request with a working version if that's the case.

### CODE REVIEW

File: extensions/typescript/package.nls.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
 	"format.insertSpaceBeforeAndAfterBinaryOperators": "Defines space handling after a binary operator.",
 	"format.insertSpaceAfterKeywordsInControlFlowStatements": "Defines space handling after keywords in a control flow statement.",
 	"format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": "Defines space handling after function keyword for anonymous functions.",
+	"format.insertSpaceBeforeFunctionParenthesis": "Defines space handling before function argument parentheses.",
```
Comment (mjbvz) [inline extensions/typescript/package.nls.json]: Please add a ` Requires TypeScript >= 2.1.5.` blurb at the end

**Review by mjbvz — APPROVED** [review]
Thank you for the contribution!

Looks good overall, let's just make sure to note which version of TypeScript is required to use this setting

**Comment by mjbvz** [comment]
Thanks! This should be in the next 1.11 insiders build

**Comment by mikhailmelnik** [comment]
Using 1.13.1 and having the opposite issue: by default VS Code adds the space and I can't find any way to remove it as otherwise it will lead to too many changes in source code. Have tried to set `"javascript.format.insertSpaceBeforeFunctionParenthesis": false` (which is `false` by default) by it doesn't help.

**Comment by mjbvz** [comment]
@mikhailmelnik Please open an issue (but also check that an extension is not causing this first)

**Pull request merged**